.github/workflows/automated-tests.yml

```diff
@@ -0,0 +1,125 @@
+name: Automated Tests
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  automated-tests:
+    name: ${{ matrix.tests.command }}
+    runs-on: ubuntu-latest
+
+    env:
+      CI: true
+      RAILS_ENV: test
+      TEST_GROUP: ${{ matrix.tests.command }}
+      CUCUMBER_RETRY: 1
+      CUCUMBER_FORMAT: Ao3Cucumber::Formatter
+
+    services:
+      database:
+        image: mariadb:10.5.4-focal
+        env:
+          MYSQL_ROOT_PASSWORD: password
+          MYSQL_DATABASE: otwarchive_test
+        ports:
+          - 3306:3306
+
+      elasticsearch:
+        image: elasticsearch:6.8.9
+        ports:
+          - 9200:9200
+
+      memcached:
+        image: memcached:1.5
+        ports:
+          - 11211:11211
+
+    strategy:
+      max-parallel: 5
```

Comment:
I forgot to ask: was there a reason you chose 5 for max-parallel, or is this something we can increase if necessary?

Reply:
It can be increased, or omitted altogether. I picked 5 because that matches the number of parallel jobs on Travis.

Reply:
Looks like we can have up to [20 concurrent jobs](https://docs.github.com/en/free-pro-team@latest/actions/reference/usage-limits-billing-and-administration#usage-limits) (across all builds - all of otwcode?) in the free plan. Let's take full advantage (#3930).